src/WebSocketHandler.js

```diff
@@ -42,7 +51,9 @@ class WebSocketHandler extends EventEmitter {
 
   endConnection() {
     // To keep compatibility with methods previously used in this singleton
-    return this.ws.endConnection();
+    if (this.ws !== null) {
+      return this.ws.endConnection();
```

Comment:
Should we set `this.ws = null`?

Reply:
Thinking more about this situation, I believe it's better. Because if we are changing the server it's important to start the new `this.ws` in the new server.

Reply:
Done. Tested with server change and everything is working fine.